### PR TITLE
[FIP-42] Add regdomadd action

### PR DESCRIPTION
**Pull request by adsorptionenthalpy**
First cut at regdomadd action for testing

Registers FIO Domain and Handle together under one action.
FIP-42 - https://github.com/fioprotocol/fips/blob/master/fip-0042.md

### CODE REVIEW

File: contracts/fio.address/fio.address.cpp [hunk 1]
```diff
@@ -809,6 +809,130 @@ namespace fioio {
...
+            fio_400_assert(domains_iter == domainsbyname.end(), "fio_name", fa.fioaddress,
+                           "Domain already registered, use regaddress instead.", ErrorDomainAlreadyRegistered);
+            
+            uint32_t domain_expiration = get_now_plus_one_year();
+            domains.emplace(actor, [&](struct domain &d) {
+                d.id = domains.available_primary_key();;
+                d.name = fa.fiodomain;
+                d.domainhash = domainHash;
+                d.expiration = domain_expiration;
+                d.account = actor.value;
```
Comment (ericbutz) [inline contracts/fio.address/fio.address.cpp:864]: owner_account_name?

File: contracts/fio.address/fio.address.cpp [hunk 2]
```diff
@@ -809,6 +809,130 @@ namespace fioio {
...
+
+            fionames.emplace(actor, [&](struct fioname &a) {
+                a.id = fionames.available_primary_key();;
+                a.name = fa.fioaddress;
+                a.addresses = pubaddresses;
+                a.namehash = string_to_uint128_hash(fa.fioaddress.c_str());;
+                a.domain = fa.fiodomain;
+                a.domainhash = domainHash;
+                a.expiration = 4294967295; //Sunday, February 7, 2106 6:28:15 AM GMT+0000 (Max 32 bit expiration)
+                a.owner_account = actor.value;
```
Comment (ericbutz) [inline contracts/fio.address/fio.address.cpp:887]: owner_account_name?

File: contracts/fio.address/fio.address.cpp [hunk 3]
```diff
@@ -809,6 +809,130 @@ namespace fioio {
...
+                d.id = domains.available_primary_key();;
+                d.name = fa.fiodomain;
+                d.domainhash = domainHash;
+                d.expiration = domain_expiration;
+                d.account = actor.value;
+            });
+
+            auto key_iter = accountmap.find(actor.value);
+
+            fio_400_assert(key_iter != accountmap.end(), "owner", to_string(actor.value),
```
Comment (ericbutz) [inline contracts/fio.address/fio.address.cpp:869]: owner_account_name?

File: contracts/fio.address/fio.address.cpp [hunk 4]
```diff
@@ -809,6 +809,130 @@ namespace fioio {
...
+
+            fio_fees(actor, asset(reg_amount, FIOSYMBOL), REGISTER_FIO_DOMAIN_ADDRESS_ENDPOINT);
+            processbucketrewards(tpid, reg_amount, get_self(), actor);
+
+            if (REGDOMADDRAM > 0) {
+                action(
+                        permission_level{SYSTEMACCOUNT, "active"_n},
+                        "eosio"_n,
+                        "incram"_n,
+                        std::make_tuple(actor, REGDOMADDRAM)
```
Comment (ericbutz) [inline contracts/fio.address/fio.address.cpp:917]: Please confirm with Pawel if it is okay to reuse REGDOMADDRAM or if this call should have its own fee.

Comment (adsorptionenthalpy) [reply contracts/fio.address/fio.address.cpp:917]: This is a new constant

File: contracts/fio.address/fio.address.cpp (hunk 3 above)
Comment (adsorptionenthalpy) [reply contracts/fio.address/fio.address.cpp:869]: I'll check on these

File: contracts/fio.address/fio.address.cpp (hunk 4 above)
Comment (ericbutz) [reply contracts/fio.address/fio.address.cpp:917]: Ah, right. looks like it is set to 5120. Looks good. Closing.

**Review by edrotthoff — COMMENTED** [review]
I think I share Eric’s comment that perhaps the owner account name should be used instead of the actor for account assignments….this is best verified with Adam and pawel.

**Comment by adsorptionenthalpy** [comment]
Updated actor based on feedback

**Review by ericbutz — APPROVED** [review]

**Review by edrotthoff — APPROVED** [review]

**Review by edrotthoff — APPROVED** [review]

**Pull request merged**